### PR TITLE
Filter tariffs cards

### PR DESCRIPTION
**Pull request by melnychenkoas**
Filter tariffs cards on the main tariffs page with real data

### CODE REVIEW

**Review by NewArtMask — APPROVED** [review]

File: src/app/ubs/ubs-admin/components/ubs-admin-tariffs/ubs-admin-tariffs-location-dashboard.component.html [hunk 1]
```diff
@@ -158,7 +163,7 @@
   </div>
 </form>
 <table class="table-location">
-  <thead *ngIf="filteredLocations?.length !== 0">
+  <thead *ngIf="cards.length !== 0">
```
Comment (hnativlyubomyr) [inline src/app/ubs/ubs-admin/components/ubs-admin-tariffs/ubs-admin-tariffs-location-dashboard.component.html:166]: *ngIf="cards.length"

File: src/app/ubs/ubs-admin/components/ubs-admin-tariffs/ubs-admin-tariffs-location-dashboard.component.ts [hunk 2]
```diff
@@ -184,13 +215,13 @@ export class UbsAdminTariffsLocationDashboardComponent implements OnInit, OnDest
     }
   }
 
-  public checkStation(item): boolean {
-    return this.selectedStation.indexOf(item) >= 0;
+  public checkSelectedItem(item, array): boolean {
```
Comment (hnativlyubomyr) [inline src/app/ubs/ubs-admin/components/ubs-admin-tariffs/ubs-admin-tariffs-location-dashboard.component.ts:218]: add params type

File: src/app/ubs/ubs-admin/components/ubs-admin-tariffs/ubs-admin-tariffs-location-dashboard.component.ts [hunk 3]
```diff
@@ -250,26 +316,35 @@ export class UbsAdminTariffsLocationDashboardComponent implements OnInit, OnDest
...
 
-  public mapCitiesInUkr(region): Array<string> {
-    return region
-      .map((element) =>
-        element.locationsDto.map((el) =>
-          el.locationTranslationDtoList.filter((it) => it.languageCode === 'ua').map((it) => it.locationName)
-        )
-      )
-      .flat(2);
+  public mapCitiesInUkr(region): Array<object> {
```
Comment (hnativlyubomyr) [inline src/app/ubs/ubs-admin/components/ubs-admin-tariffs/ubs-admin-tariffs-location-dashboard.component.ts:333]: region - string[] -?

File: src/app/ubs/ubs-admin/components/ubs-admin-tariffs/ubs-admin-tariffs-location-dashboard.component.ts [hunk 4]
```diff
@@ -322,13 +399,29 @@ export class UbsAdminTariffsLocationDashboardComponent implements OnInit, OnDest
 
   checkRegionValue(value): void {
     let currentRegion;
-    if (value === 'Усі') {
+    if (value === 'Усі' || value === '') {
```
Comment (hnativlyubomyr) [inline src/app/ubs/ubs-admin/components/ubs-admin-tariffs/ubs-admin-tariffs-location-dashboard.component.ts:402]: if (value === 'Усі' || !value)

**Review by hnativlyubomyr — APPROVED** [review]

**Review by Myh45 — APPROVED** [review]

**Review by Tvido — APPROVED** [review]

**Review by voronkooleksandr — APPROVED** [review]

**Pull request merged**